vowpalwabbit/parse_args.cc

```diff
@@ -1003,6 +1006,13 @@ void parse_reductions(vw& all)
   all.reduction_stack.push_back(bs_setup);
 
   all.l = setup_base(all);
+
+  if (all.probabilities)
+  { if (!all.vm.count("oaa") && !all.vm.count("csoaa_ldf"))
+      THROW("--probabilities can only be used with --oaa=N or --csoaa_ldf=mc");
+    if (!all.vm.count("loss_function") || all.vm["loss_function"].as<string>() != "logistic" )
```

Comment:
Yes, more modular is preferred.  It may be impossible to remove 
all.probabilities entirely, but you should not need to do more than 
assign to it in (cs)oaa.  Instead, use a local version of the variable.  
This will make things easier when I get some time to refactor the code 
to remove all.probabilities later and deal with other label type issues 
later.

-John

On 10/28/2015 12:47 PM, Martin Popel wrote:
>
> In vowpalwabbit/parse_args.cc 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/837#discussion_r43283119>:
>
> > @@ -1003,6 +1006,13 @@ void parse_reductions(vw& all)
> >    all.reduction_stack.push_back(bs_setup);
> >
> >    all.l = setup_base(all);
> > +
> > +  if (all.probabilities)
> > +  { if (!all.vm.count("oaa") && !all.vm.count("csoaa_ldf"))
> > +      THROW("--probabilities can only be used with --oaa=N or --csoaa_ldf=mc");
> > +    if (!all.vm.count("loss_function") || all.vm["loss_function"].as<string>() != "logistic" )
>
> As I wrote in #774 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/774>: I can move 
> the |--probabilities| option to |oaa.cc| and |csoaa.cc| (i.e. 
> duplicate its definition), but probably I will still need to store it 
> inside |all| (similarly, e.q. |csoaa_rank| aka |multilabel_prediction| 
> is stored inside |all| although it is valid only for some reductions). 
> The problem would be that |vw -h| would not list the |--probabilities| 
> options (see #665 
> <https://github.com/JohnLangford/vowpal_wabbit/issues/665>).
> So should I really move the option |--probabilities| to oaa.cc and 
> csoaa.cc?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/837/files#r43283119>.
>